tools/clang/test/CodeGenSPIRV/bezier.domain.hlsl2spv

```diff
@@ -0,0 +1,197 @@
+// OpDecorate %in_var_TANVCORNER Location 3
+// OpDecorate %in_var_TANWEIGHTS Location 4
+// OpDecorate %in_var_hullShaderOutput Location 5
+// OpDecorate %out_var_NORMAL Location 0
+// OpDecorate %out_var_TEXCOORD Location 1
+// OpDecorate %out_var_TANGENT Location 2
+// OpDecorate %out_var_BITANGENT Location 3
+// %uint = OpTypeInt 32 0
+// %void = OpTypeVoid
+// %3 = OpTypeFunction %void
+// %float = OpTypeFloat 32
+// %uint_4 = OpConstant %uint 4
+// %_arr_float_uint_4 = OpTypeArray %float %uint_4
+// %uint_2 = OpConstant %uint 2
+// %_arr_float_uint_2 = OpTypeArray %float %uint_2
+// %v3float = OpTypeVector %float 3
+// %_arr_v3float_uint_4 = OpTypeArray %v3float %uint_4
+// %v2float = OpTypeVector %float 2
+// %_arr_v2float_uint_4 = OpTypeArray %v2float %uint_4
+// %v4float = OpTypeVector %float 4
+// %HS_CONSTANT_DATA_OUTPUT = OpTypeStruct %_arr_float_uint_4 %_arr_float_uint_2 %_arr_v3float_uint_4 %_arr_v2float_uint_4 %_arr_v3float_uint_4 %_arr_v3float_uint_4 %v4float
+// %_ptr_Function_HS_CONSTANT_DATA_OUTPUT = OpTypePointer Function %HS_CONSTANT_DATA_OUTPUT
+// %_ptr_Input__arr_float_uint_4 = OpTypePointer Input %_arr_float_uint_4
+// %_ptr_Input__arr_float_uint_2 = OpTypePointer Input %_arr_float_uint_2
+// %_ptr_Input__arr_v3float_uint_4 = OpTypePointer Input %_arr_v3float_uint_4
+// %_ptr_Input__arr_v2float_uint_4 = OpTypePointer Input %_arr_v2float_uint_4
+// %_ptr_Input_v4float = OpTypePointer Input %v4float
+// %_ptr_Function_v2float = OpTypePointer Function %v2float
+// %_ptr_Input_v2float = OpTypePointer Input %v2float
+// %BEZIER_CONTROL_POINT = OpTypeStruct %v3float
+// %uint_16 = OpConstant %uint 16
+// %_arr_BEZIER_CONTROL_POINT_uint_16 = OpTypeArray %BEZIER_CONTROL_POINT %uint_16
+// %_ptr_Function__arr_BEZIER_CONTROL_POINT_uint_16 = OpTypePointer Function %_arr_BEZIER_CONTROL_POINT_uint_16
+// %_ptr_Input__arr_BEZIER_CONTROL_POINT_uint_16 = OpTypePointer Input %_arr_BEZIER_CONTROL_POINT_uint_16
+// %DS_OUTPUT = OpTypeStruct %v3float %v2float %v3float %v3float %v4float
+// %_ptr_Output_v3float = OpTypePointer Output %v3float
+// %_ptr_Output_v2float = OpTypePointer Output %v2float
+// %_ptr_Output_v4float = OpTypePointer Output %v4float
+// %68 = OpTypeFunction %DS_OUTPUT %_ptr_Function_HS_CONSTANT_DATA_OUTPUT %_ptr_Function_v2float %_ptr_Function__arr_BEZIER_CONTROL_POINT_uint_16
+// %_ptr_Function_DS_OUTPUT = OpTypePointer Function %DS_OUTPUT
+// %gl_TessLevelOuter = OpVariable %_ptr_Input__arr_float_uint_4 Input
+// %gl_TessLevelInner = OpVariable %_ptr_Input__arr_float_uint_2 Input
+// %in_var_TANGENT = OpVariable %_ptr_Input__arr_v3float_uint_4 Input
+// %in_var_TEXCOORD = OpVariable %_ptr_Input__arr_v2float_uint_4 Input
+// %in_var_TANUCORNER = OpVariable %_ptr_Input__arr_v3float_uint_4 Input
+// %in_var_TANVCORNER = OpVariable %_ptr_Input__arr_v3float_uint_4 Input
+// %in_var_TANWEIGHTS = OpVariable %_ptr_Input_v4float Input
+// %gl_TessCoord = OpVariable %_ptr_Input_v2float Input
```

Comment:
According to [Vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.0-wsi_extensions/html/vkspec.html#interfaces-builtin-variables): "The variable decorated with TessCoord must be declared as three-component vector of 32-bit floating-point values."

Reply:
So we may need to emit additional code to read the first two components of `TessCoord` and make a new argument out of them.

Reply:
Good catch! it seems that HLSL has the 2-component position, whereas the TessCoord builtin in SPIR-V is a 3-component vector. I should extract the first 2 components in the wrapper function.